models/enterprise_sonic/snmp/sonic_snmp.yml

```diff
@@ -1,53 +1,253 @@
+                  - Name of community group
+                type: str
+              name:
+                descripton:
+                  - Name of community
+                type: str
+                required: true
+          engine:
+            description:
+              - SNMP engine ID
+            type: str
+          user:
+            description:
+              - List of SNMP user configuration
+            type: list
+            elements: dict
+            suboption:
+              group:
+                description:
+                  - Name of user group
+                type: str
+              name:
+                description:
+                  - Name of user
+                type: str
+                required: true
+              auth:
+                description:
+                  - User authentication configuration
+                type: dict
+                suboptions:
+                  md5_key: 
+                    description: 
+                      - MD5 authentication key
+                    type: str
+                  sha_key:
+                    description:
+                      - SHA authentication key
+                    type: str
+              priv:
+                description:
+                  - Privacy configuration
+                type: dict
+                suboptions:
+                  des_key:
+                    description:
+                      - DES ecryption
+                    type: str
+                  aes_key:
+                    description:
+                      - AES encryption
+                    type: str
+              encrypted:
+                description:
+                  - Enable/disable encryption
+                type: bool       
+          view:
+            description:
+             - List of SNMP view configuration
+            type: list
+            elements: dict
+            suboptions:
+              name:
+                description:
+                  - Name of the view
+                type: str
+                required: true
+              include:
+                description:
+                  - List of subtrees included in the view
+                type: list
+                elements: str
+              exclude:
+                description:
+                  - List of subtrees excluded from the view
+                type: list
+                elements: str
+          contact:
+            description:
+              - SNMP server contact information
+            type: str
+          location:
+            description:
+              -SNMP server location information
+            type: str
+          enable_trap:
+            description:
+              - Specify trap(s) to enable
+            type: list
+            elements: str
+            choices:
+              - auth-fail
+              - bgp
+              - config-change
+              - link-down
+              - link-up
+              - ospf
+              - all
```

Comment:
Please specify the behavior for 'all' since it overrides other choices (and is mutually exclusive).